base_layer/core/src/transactions/transaction_components/transaction_output.rs

```diff
@@ -291,14 +291,38 @@ impl TransactionOutput {
         encrypted_value: &EncryptedValue,
         minimum_value_promise: MicroTari,
     ) -> [u8; 32] {
+        let message = TransactionOutput::build_metadata_signature_message(
+            version,
+            script,
+            features,
+            covenant,
+            encrypted_value,
+            minimum_value_promise,
+        );
         let common = DomainSeparatedConsensusHasher::<TransactionHashDomain>::new("metadata_signature")
             .chain(&version)
             .chain(ephemeral_pubkey)
             .chain(ephemeral_commitment)
-            .chain(script)
-            .chain(features)
             .chain(sender_offset_public_key)
             .chain(commitment)
+            .chain(&message);
+        match version {
+            TransactionOutputVersion::V0 | TransactionOutputVersion::V1 => common.finalize(),
+        }
+    }
+
+    /// Convenience function that calculates the message for the metadata commitment signature
+    pub fn build_metadata_signature_message(
```

Comment:
Can you document why these particular fields are included in this hash, but not others?